ops/model.py

```diff
@@ -279,7 +290,12 @@ def __init__(self, name, relation_id, backend):
     @property
     def network(self):
         if self._network is None:
-            self._network = Network(self._backend.network_get(self.name, self._relation_id))
+            try:
+                self._network = Network(self._backend.network_get(self.name, self._relation_id))
+            except RelationNotFoundError:
+                # If a relation is dead, we can still get network info associated with an endpoint itself
+                # not associated with a particular relation id.
+                self._network = Network(self._backend.network_get(self.name))
```

Comment:
We might want to actually do:

if self._relation_id is not None:
  self._network = ...

Because otherwise it calls with the same args twice.

Reply:
https://github.com/canonical/operator/pull/157/commits/a9e15b0059086c7c3fe20af4af18e281cb70ad30

Reply:
This seems a bit verbose vs:
```
            try:
                self._network = Network(self._backend.network_get(self.name, self._relation_id))
            except RelationNotFoundError:
                if self._relation_id is not None:
                   # If a relation is dead, we can still get network info associated with an endpoint itself
                   self._network = Network(self._backend.network_get(self.name))
```

Reply:
that's a different behaviour though, what's there now is more like
```python
            try:
                self._network = Network(self._backend.network_get(self.name, self._relation_id))
            except RelationNotFoundError:
                if self._relation_id is None:
                    raise
                # If a relation is dead, we can still get network info associated with an endpoint itself
                self._network = Network(self._backend.network_get(self.name))
```

Reply:
https://github.com/canonical/operator/pull/157/commits/d806429c7daeddd6d0d3c45f8646698fc015c760